### ISSUE TITLE
SlurmProvider mem_per_node passes incorrectly formatted memory string

### ISSUE THREAD

**Issue by mattwelborn** [Parsl/parsl#1416]
**Describe the bug**
When specifying the mem_per_node option to SlurmProvider, the submission script requests memory with "#SBATCH --mem=120.0g" (rather than "#SBATCH --mem=120g"), which is invalid in SLURM version 17.11.8. The error in the parsl log is:

```
2019-10-29 17:26:10.215 parsl.providers.slurm.slurm:185 [DEBUG]  Requesting one block with 1 nodes
2019-10-29 17:26:10.215 parsl.providers.slurm.slurm:202 [DEBUG]  Writing submit script
2019-10-29 17:26:10.218 parsl.providers.slurm.slurm:206 [DEBUG]  moving files
2019-10-29 17:26:10.244 parsl.providers.slurm.slurm:222 [ERROR]  Retcode:1 STDOUT: STDERR:sbatch: error: invalid memory constraint 120.0g
2019-10-29 17:26:10.244 parsl.executors.high_throughput.executor:569 [DEBUG]  Launched block 0->None
2019-10-29 17:26:10.244 parsl.dataflow.flow_control:142 [ERROR]  Flow control callback threw an exception - logging and proceeding anyway
Traceback (most recent call last):
  File "/home/mwelborn/miniconda3/envs/parsl9_dev/lib/python3.6/site-packages/parsl/dataflow/flow_control.py", line 140, in make_callback
    self.callback(tasks=self._event_buffer, kind=kind)
  File "/home/mwelborn/miniconda3/envs/parsl9_dev/lib/python3.6/site-packages/parsl/dataflow/strategy.py", line 260, in _strategy_simple
    executor.scale_out(excess_blocks)
  File "/home/mwelborn/miniconda3/envs/parsl9_dev/lib/python3.6/site-packages/parsl/executors/high_throughput/executor.py", line 572, in scale_out
    "Attempts to provision nodes via provider has failed"))
parsl.executors.errors.ScalingFailed: ('slurm', 'Attempts to provision nodes via provider has failed'
```

**To Reproduce**
Steps to reproduce the behavior, for e.g:
1. Setup Parsl 0.8.0 with Python 3.6 on a cluster with slurm 17.11.8
2. Create a SlurmProvider with mem_per_node specified
3. Create tasks

**Expected behavior**
Parsl should complete work without incident. 

**Environment**
 - OS: Linux x86_64
 - Python 3.6.7
 - Parsl 0.9.0
 - SLURM 17.11.8

**Distributed Environment**
- Where are you running the Parsl script from ? Login node
- Where do you need the workers to run ? Compute nodes

**Comment by yadudoc** [comment]
@mattwelborn Could you share the config you are using please ?

**Comment by mattwelborn** [comment]
Config below. Also attached is the runinfo. 


```
Config(
    app_cache=True,
    checkpoint_files=None,
    checkpoint_mode=None,
    checkpoint_period=None,
    data_management_max_threads=10,
    executors=[HighThroughputExecutor(
        address='calogin1',
        cores_per_worker=4,
        heartbeat_period=30,
        heartbeat_threshold=120,
        interchange_port_range=(55000, 56000),
        label='QCFractal_Parsl_Slurm_Executor',
        launch_cmd='process_worker_pool.py {debug} {max_workers} -p {prefetch_capacity} -c {cores_per_worker} -m {mem_pe
r_worker} --poll {poll_period} --task_url={task_url} --result_url={result_url} --logdir={logdir} --block_id={{block_id}}
 --hb_period={heartbeat_period} --hb_threshold={heartbeat_threshold} ',
        managed=True,
        max_workers=192,
        mem_per_worker=None,
        poll_period=10,
        prefetch_capacity=0,
        provider=SlurmProvider(
            'normal_q',
            channel=LocalChannel(
                envs={},
                script_dir=None,
                userhome='/home/mwelborn/qcarchive/managers/cascades_wd'
            ),
            cmd_timeout=120,
            cores_per_node=32,
            exclusive=True,
            init_blocks=0,
            launcher=SingleNodeLauncher(),
            max_blocks=24,
            mem_per_node=120.0,
            min_blocks=0,
            move_files=True,
            nodes_per_block=1,
            parallelism=1,
            scheduler_options='#SBATCH -A themolssi\n\n#SBATCH --exclusive\n',
            walltime='71:00:00',
            worker_init='source /home/mwelborn/qcarchive/boot_qcarchive_env.sh\n'
        ),
        storage_access=None,
        suppress_failure=True,
        worker_debug=False,
        worker_logdir_root=None,
        worker_port_range=(54000, 55000),
        worker_ports=None,
        working_dir=None
    )],
    initialize_logging=True,
    lazy_errors=True,
    max_idletime=120.0,
    monitoring=None,
    retries=0,
    run_dir='runinfo',
    strategy='simple',
    usage_tracking=False
)
```
[runinfo_1416.tar.gz](https://github.com/Parsl/parsl/files/3790397/runinfo_1416.tar.gz)

**Comment by yadudoc** [comment]
@mattwelborn This should now be fixed in master with #1423 merged.
Let's reopen if you think this is still an issue.
